servers/physics_3d/godot_collision_solver_3d_sat.cpp

```diff
@@ -927,7 +936,7 @@ static void analytic_sphere_cylinder_collision(real_t p_radius_a, real_t p_radiu
 	}
 	Vector3 point_a = p_transform_a.origin + (p_radius_a + p_margin_a) * axis;
 	Vector3 point_b = (withMargin ? nearest + p_margin_b * axis : nearest);
```

Comment:
There's an identical problem here:
 1. The axis points from A to B where it should point from B to A
 2. The margins are again applied in the same direction rather than away from the objects.

Fixing this one fixes a second regression in https://github.com/fabriceci/Godot-Physics-Tests.